### PR TITLE
Overrided equals & hashcode for Rectangle class

### PR DESCRIPTION
**Pull request by andriylagun**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (!(object instanceof Rectangle) || object == null) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: what the difference between instanceof and getClass()?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
...
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
+        return (length == null || rectangle.length == null)
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
        return length == null
```
this should be enough

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (andriylagun) [reply src/main/java/core/basesyntax/Rectangle.java]: I see my mistake, if i will create subclass from Rectangle, object of this subclass will not equals to Rectangle object and Rectangle will be equal to child-class object. For strict type matching I need to use getClass()

**Review by dmytro-hryhoruk — COMMENTED** [review]
1) don't call your commit "Some fixes".Let's give descriptive names to commit;
2) fix travis;

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
...
+                && (color == null || rectangle.color == null)
+                ? color == rectangle.color : color.equals(rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 51;
+        result = 31 * result + (getLength() != null ? getLength().hashCode() : 0);
+        result = 31 * result + (getWidth() != null ? getWidth().hashCode() : 0);
+        result = 31 * result + (getColor() != null ? getColor().hashCode() : 0);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java:53]: should we use getters inside class?

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**